_gsocblogs/2024/blog_Key4hep_BraulioRivas.md

```diff
@@ -0,0 +1,131 @@
+| :----------------------------------------- | :--------------------------------------------: | :------------------------------------------------------------------------------------------------------------------------------ |
+| Visualization of MCParticle + RecoParticle | [#43](https://github.com/key4hep/eede/pull/43) | Views for `RecoParticle` and existing relations between this and `MCParticle`.                                                  |
+| Views for cluster and track                | [#48](https://github.com/key4hep/eede/pull/48) | I added a views for `Cluster` and `Track` plus relations that exists between `MCParticles`-`Cluster` and `MCParticles`-`Track`. |
+| Initial views for particleID and vertex    | [#55](https://github.com/key4hep/eede/pull/55) | More views for `ParticleId` and `Vertex`.                                                                                       |
+
+**Pixi**
+
+eede is a static website. So we are using plain Javascript, HTML and CSS. But for visualization, we were using the Canvas API,
+so all datatypes are drawn in a white canvas. Nevertheless, when having many objects in a same view, it was almost impossible
+to scroll or drag objects around. So I [analyzed](https://github.com/key4hep/eede/pull/63) some of the weak points at that moment.
+I came to the conclusion to change Canvas API to some more performant tool, and to find a way to render objects more efficiently.
+After some research, I found [pixi.js](https://pixijs.com/), a library (which has a minified `js` version) that uses WebGL
+to render any kind of content on a canvas very, but very fast. After a bit of rework, I finally transferred everything to pixi.
+The difference was big. Before, eede struggled to render a tree of `MCParticle` from this file https://fccsw.web.cern.ch/fccsw/eede/wzp6_ee_mumuH_ecm240_CLD_RECO.edm4hep.json
+(by some reason, it was only possible in Firefox, but not on a chromium based browser). But it later had no problem, it
+rendered everything fast without crashing.
+Other very noticeable example is when rendering $15000+$ `MCParticle`s. When using Canvas API, the ram consumption went up by a lot
+and freezed the computer. But after using pixi and applying some techniques, like [culling](https://pixijs.com/8.x/guides/basics/scene-graph#culling)
+it had no problem and takes only the necessary amount of memory.
+| Pull Request| PR Number | Description |
+| :--- | :----: | :--- |
+| Use pixi | [#65](https://github.com/key4hep/eede/pull/65) | I changed the rendering engine from Canvas API to pixi.js. This PR also includes a few changes to make the rendering more efficient. |
+
+**Filters**
+
+One of the last tasks was to develop a set of filtering functionalities, because in one single event may exists lots
+of objects, making it harder to visualize. So, I developed filters for every datatype, that would allow to filter
```

Comment:
```suggestion
One of the last tasks was to develop a set of filtering functionalities, because in one single event many
objects might exist, making it harder to visualize. So, I developed filters for every datatype, that allow to filter
```